packages/svelte/src/compiler/phases/3-transform/client/visitors/javascript-runes.js

```diff
@@ -27,35 +65,10 @@ export const javascript_visitors_runes = {
 				const is_private = type === 'PrivateIdentifier';
 				if (is_private) private_ids.push(name);
 
-				if (definition.value?.type === 'CallExpression') {
-					const rune = get_rune(definition.value, state.scope);
-					if (
-						rune === '$state' ||
-						rune === '$state.frozen' ||
-						rune === '$derived' ||
-						rune === '$derived.by'
-					) {
-						/** @type {import('../types.js').StateField} */
-						const field = {
-							kind:
-								rune === '$state'
-									? 'state'
-									: rune === '$state.frozen'
-										? 'frozen_state'
-										: rune === '$derived.by'
-											? 'derived_call'
-											: 'derived',
-							// @ts-expect-error this is set in the next pass
-							id: is_private ? definition.key : null
-						};
-
-						if (is_private) {
-							private_state.set(name, field);
-						} else {
-							public_state.set(name, field);
-						}
-					}
-				}
+				create_state_field(definition, is_private, name);
+			} else if (definition.type === 'PropertyDefinition' && definition.key.type === 'Literal') {
+				const name = definition.key.value?.toString().replace('-', '_');
```

Comment:
Don't think this is enough - there could be all sorts of whacky characters in there, better to use a "non-valid ID identifier" regex (I believe there exists one already)

Reply:
Oh great, Ill update soon

Reply:
Also just realized I need `replaceAll`

Reply:
@dummdidumm fixed it in the new commit...should i add more tests for all the same thing that were tested with normal identifiers?